### PR TITLE
Upgrade analyzer to 5.2.0 and fix deprecated warnings

### PR DESCRIPTION
**Pull request by rrousselGit**
The CI is currently failing due to the new 5.2.0 version of analyzer deprecating a bunch of methods.

### CODE REVIEW

File: json_annotation/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 4.7.1-dev+1
+
+- Require Dart SDK `>=2.18.0`.
```
Comment (kevmoo) [inline json_annotation/CHANGELOG.md:3]: 4.7.1 hasn't been shipped yet (hence the -dev) – so you can put this in the blew section

File: json_annotation/pubspec.yaml [hunk 2]
```diff
@@ -1,12 +1,12 @@
 name: json_annotation
-version: 4.7.1-dev
+version: 4.7.2-dev+1
```
Comment (kevmoo) [inline json_annotation/pubspec.yaml:2]: revert this

File: json_serializable/pubspec.yaml [hunk 3]
```diff
@@ -1,14 +1,14 @@
 name: json_serializable
-version: 6.5.4
+version: 6.6.0-dev
```
Comment (kevmoo) [inline json_serializable/pubspec.yaml:2]: 6.5.5-dev is better. No new features – just bumping the SDK constraint

File: checked_yaml/pubspec.yaml [hunk 4]
```diff
@@ -6,7 +6,7 @@ description: >-
   package:json_serializable and package:yaml.
 repository: https://github.com/google/json_serializable.dart/tree/master/checked_yaml
 environment:
-  sdk: '>=2.17.0 <3.0.0'
+  sdk: '>=2.18.0 <3.0.0'
```
Comment (kevmoo) [inline checked_yaml/pubspec.yaml:9]: need a changelog update here ,too!

**Review by kevmoo — CHANGES_REQUESTED** [review]

**Comment by rrousselGit** [comment]
Done, I've updated them all

**Review by kevmoo — APPROVED** [review]

**Pull request merged**